File: my-folder/3213-count-subarrays-where-max-element-appears-at-least-k-times/solution.py
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        n = len(nums)
        ans = 0
        m = max(nums)

        left, right = 0, 0
        table = defaultdict(int)
        while right < n:
            if nums[right] == m:
                table[m]+=1
            while left <= right and table[m] >= k:
                ans+=1
                right_count = n - right - 1
                ans+=right_count
                if nums[left] == m:
                    table[m]-=1
                left+=1
            right+=1
        return ans

File: my-folder/3213-count-subarrays-where-max-element-appears-at-least-k-times/solution.cpp
class Solution {
public:
    long long countSubarrays(vector<int>& nums, int k) {
        int n = nums.size();
        int max = *max_element(nums.begin(), nums.end());
        int count = 0;
        long long ans = 0;


        int left = 0;
        int right = 0;

        while (right < n){
            if (nums[right] == max){
                count+=1;
            }
            while (left <= right && count>=k){
                ans+=1;
                int right_count = n - right - 1;
                ans+=right_count;
                if (nums[left] == max){
                    count-=1;
                }
                left+=1;
            } 
            right+=1;
        }
        return ans;
    }
};
